02_bubble/jonathanheaden/main_test.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestMainOutput(t *testing.T) {
+	// Given
+	r := require.New(t)
+	var buf bytes.Buffer
+	out = &buf
+
+	// When
+	main()
+
+	// Then
+	expected := strconv.Quote("[1 2 3 5]\n")
+	actual := strconv.Quote(buf.String())
+	r.Equalf(expected, actual, "Unexpected output in main()")
+}
+
+func TestNegativesOutput(t *testing.T) {
+	// Given
+	r := require.New(t)
+	var buf bytes.Buffer
+	out = &buf
+
+	// When
+	fmt.Fprintln(out, bubble([]int{3, -2, 1, -5, 7}))
+
+	// Then
+	expected := strconv.Quote("[-5 -2 1 3 7]\n")
+	actual := strconv.Quote(buf.String())
+	r.Equalf(expected, actual, "Unexpected output in main()")
+}
+
+func TestAllElemetsEqual(t *testing.T) {
```

Comment:
Nice work, well structured tests. You should (and probably have) look to structure your tests so you are iterating over different cases in the future labs. That way you avoid the repetitive nature of the test skeleton code.

Reply:
will do
thanks